File: client/frontend/src/types/task.ts
export enum TaskStatus {
    PENDING = 'PENDING',
    COMPLETED = 'COMPLETED',
    FOLLOW_UP = 'FOLLOW_UP',
    CANCELLED = 'CANCELLED'
}

export const TaskStatusText = {
    [TaskStatus.PENDING]: '待完成',
    [TaskStatus.COMPLETED]: '已完成',
    [TaskStatus.FOLLOW_UP]: '待跟进',
    [TaskStatus.CANCELLED]: '已取消'
}

export enum TimeStatus {
    NOT_STARTED = 'NOT_STARTED',
    IN_PROGRESS = 'IN_PROGRESS',
    OVERDUE = 'OVERDUE',
    COMPLETED = 'COMPLETED'
}

export const TimeStatusText = {
    [TimeStatus.NOT_STARTED]: '未开始',
    [TimeStatus.IN_PROGRESS]: '进行中',
    [TimeStatus.OVERDUE]: '已超时',
    [TimeStatus.COMPLETED]: '已完成'
}

export enum Priority {
    LOW = 1,
    MEDIUM = 2,
    HIGH = 3
}

export const PriorityText = {
    [Priority.LOW]: '低',
    [Priority.MEDIUM]: '中',
    [Priority.HIGH]: '高'
}

export interface Task {
    id: string;                    // 任务ID
    title: string;                  // 任务名称
    description: string;            // 任务描述
    priority: 'low' | 'medium' | 'high';  // 优先级
    estimatedDuration: number;      // 预计时间（分钟）
    deadline: Date;                 // 截止时间
    status: 'pending' | 'inProgress' | 'completed';  // 任务状态
    createdAt: Date;                // 创建时间
    updatedAt: Date;                // 更新时间
}

export type TaskPriority = Task['priority']
export type TaskStatus = Task['status']

export interface TaskFormData {
    name: string;
    description?: string;
    priority: Priority;
    startTime?: Date;
    endTime?: Date;
    status: TaskStatus;
    timeStatus: TimeStatus;
    type?: string;
    cycle?: string;
    estimatedTime?: number;
} 

File: client/backend/app/models/task.py
from sqlalchemy import Column, Integer, String, DateTime, Boolean, ForeignKey, Enum
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
import enum
from ..db.base_class import Base

class TaskStatus(str, enum.Enum):
    PENDING = "pending"
    COMPLETED = "completed"
    FOLLOW_UP = "follow_up"
    CANCELLED = "cancelled"

class TimeStatus(str, enum.Enum):
    NOT_STARTED = "not_started"
    IN_PROGRESS = "in_progress"
    OVERDUE = "overdue"
    COMPLETED = "completed"

class TaskType(str, enum.Enum):
    WORK = "work"
    STUDY = "study"
    LIFE = "life"
    OTHER = "other"

class Task(Base):
    __tablename__ = "tasks"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    description = Column(String, nullable=True)
    priority = Column(Integer, nullable=False)
    start_time = Column(DateTime(timezone=True), server_default=func.now())
    end_time = Column(DateTime(timezone=True), nullable=True)
    status = Column(Enum(TaskStatus), default=TaskStatus.PENDING)
    time_status = Column(Enum(TimeStatus), default=TimeStatus.NOT_STARTED)
    task_type = Column(Enum(TaskType), nullable=True)
    is_locked = Column(Boolean, default=False)
    estimated_time = Column(Integer, nullable=True)  # in minutes
    location = Column(String, nullable=True)
    commute_time = Column(Integer, nullable=True)  # in minutes
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    # 周期性任务相关字段
    is_periodic = Column(Boolean, default=False)
    period_type = Column(String, nullable=True)  # daily, weekly, monthly
    period_value = Column(Integer, nullable=True)  # 周期值
    period_days = Column(String, nullable=True)  # 用于存储每周的特定日期，如"1,3,5"表示周一、周三、周五

    # 任务计数
    total_count = Column(Integer, default=1)
    completed_count = Column(Integer, default=0) 